### ISSUE TITLE
What changes did you make in the skimage code

### ISSUE THREAD

**Issue by myalos** [SJTU-ViSYS/StructDepth#8]
Thanks for sharing code!
What changes did you make in the skimage code，can you release the changing code and explain why ?

**Comment by Cresynia** [comment]
Hello!I met a problem and want to ask if you have ever met it before.I made some changes to the network, and at first it can run successfully,but after several epochs,the problem occur:
![0DI RZKTGRG6_0J9~C77Y%B](https://user-images.githubusercontent.com/90834089/235556983-f869acfd-5495-499b-8540-b779922074fa.png)

**Comment by myalos** [comment]
no, pred_depth is generated by  _, depth = disp_to_depth(disp, self.opt.min_depth, self.opt.max_depth)
maybe you could check this and then continue to check according to the function call relationship.

**Comment by Cresynia** [comment]
OK,thanks!I'll try it.And I want to ask when I haven't make changes to the network,this problem didn't occur, and after I made changes,it occur,can it explain the problem is in the network?

**Comment by myalos** [comment]
yes

**Comment by Cresynia** [comment]
OK!

**Comment by Cresynia** [comment]
I think maybe I found the reasons, it's the GPU memory problem,I can continue trainig on the previous model.And I want to ask is it normal that the accuracy decrease sharply for the first several epochs after I add network to the original network?

**Comment by myalos** [comment]
i think yes, since it is normal that adding a not compatible network at first.

**Comment by Cresynia** [comment]
Thanks！It becomes better now!😃



---Original---
From: ***@***.***&gt;
Date: Wed, May 3, 2023 07:59 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [SJTU-ViSYS/StructDepth] What changes did you make in the skimagecode (Issue #8)




 
i think yes, since it is normal that adding a not compatible network at first.

**Comment by Cresynia** [comment]
Hello!I want to ask you some questions about structdepth.I want to use it to be my baseline(without the losses),and I don't know if the result of it is the DSO one in this picture?
![image](https://user-images.githubusercontent.com/90834089/236665048-d461220f-8c06-4f7d-a83f-74bddb1b22ae.png)

**Comment by Cresynia** [comment]
And when I run structdepth with the pretrained model on the Imagenet, the problems occurred:
![image](https://user-images.githubusercontent.com/90834089/236665739-c638f907-7b9e-4355-8c16-b47f3b66cd6f.png)
it didn't change!So strange.At first,I used the P2net's model as the structdepth do as pretrained model, the result is normal,because I want to use the one without structdepth's losses as my baseline,I want to change the pretrained model(use the Imagenet one).Have you ever meet this problems?

**Comment by myalos** [comment]
no, i haven't. i think that maybe training structdepth needs a good depth network at first. That means, if you change the pretrained model, the depth network will be far worse than p2net.

**Comment by Cresynia** [comment]
Yes,it needs the good network  when using the norm loss and the planar loss,but if I don't use the two losses at first,and change the network,training a good network and use losses later,is that ok?I do this,and the problem occurred.Thanks for your reply!







                       
Original Email
                       
                     

Sender:"goon"< ***@***.*** &gt;;

Sent Time:2023/5/8 21:33

To:"SJTU-ViSYS/StructDepth"< ***@***.*** &gt;;

Cc recipient:"Cresynia"< ***@***.*** &gt;;"Comment"< ***@***.*** &gt;;

Subject:Re: [SJTU-ViSYS/StructDepth] What changes did you make in the skimagecode (Issue #8)




 
no, i haven't. i think that training structdepth needs a good depth network at first. If you change the pretrained model, the depth network is far worse than p2net.
 
&mdash;

**Comment by myalos** [comment]
I didn't try it. I think it could work.

**Comment by myalos** [comment]
I didn't try it. I think it could work.

**Comment by Cresynia** [comment]
Ok,thanks!I want to ask the p2net without its planar loss is the same as the struct depth without its Manhattan norm loss and planar loss?



---Original---
From: ***@***.***&gt;
Date: Tue, May 9, 2023 18:11 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [SJTU-ViSYS/StructDepth] What changes did you make in the skimagecode (Issue #8)




 
I didn't try it. I think it could work.

**Comment by myalos** [comment]
Sorry for late, if "same" refer to accuracy, maybe not the same, since batch size, training set, pretrained model are different. If we don't take these into account, i think they are the same, both use same network structure and training method, both sparse rgb points.

**Comment by Cresynia** [comment]
OK！Thanks！



---Original---
From: ***@***.***&gt;
Date: Fri, May 12, 2023 16:25 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [SJTU-ViSYS/StructDepth] What changes did you make in the skimagecode (Issue #8)




 
Sorry for late, if "same" refer to accuracy, maybe not the same, since batch size, training set, pretrained model are different. If we don't take these into account, i think they are the same, both use same network structure and training method, both sparse rgb points.

**Comment by Cresynia** [comment]
Hello! I want to know how to get the gound truth image of the picture in the NYUv2 dataset, I'm not clear about it, could you give some idea? Thanks!

**Comment by myalos** [comment]
This was downloaded a long time ago, it should be in the evaluation section of this website https://github.com/svip-lab/Indoor-SfMLearner

**Comment by Cresynia** [comment]
Thanks! I'll see it.

**Comment by Cresynia** [comment]
Thanks so much !!!!!! I can do it !!! Thanks !!!! 😊

**Comment by myalos** [comment]
happy to hear that
